src/components/editor/TextArea.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import RIEStatefulBase from 'riek/lib/RIEStatefulBase';
+export default class TextArea extends RIEStatefulBase {
+    keyDown = (event) => {
+        if (event.keyCode === 27) { this.cancelEditing() }     // Escape
+    };
+
+    renderEditingComponent = () => {
+        return <textarea
+            rows={this.props.rows}
+            cols={this.props.cols}
+            disabled={this.state.loading}
+            className={this.makeClassString()}
+            defaultValue={this.props.value}
+            onInput={this.textChanged}
+            onBlur={this.finishEditing}
+            ref="input"
+            onKeyDown={this.keyDown}
+            {...this.props.editProps} />;
+    };
+
+    renderNormalComponent = () => {
+        const value = this.state.newValue || this.props.value
+        const spans_and_brs = []
+        var anySpan = false
+        let i = 0
+        value.split("\n").map(line => {
```

Comment:
This isn't really the right use of a `map`.  If you aren't taking the output of the map, and instead are using it to iterate through the array, use the `forEach`.  I understand why you ended up doing this, as there isn't a 1-1 relationship between input and output elements in the array you are building.  If you really wanted to use `map`, you could have the output here be an array (that may have 1 or 2 elements), and chain `flat()` to it.  I'm personally ok with just using `forEach` though.

Reply:
I'm thinking about keeping the map, and using it to return the length of all the lines to see if there are only empty lines:

`const lengths = value.split("\n").map(line => {
            spansAndBrs.push(<div key={spansAndBrs.length}>{line}</div>)
            return line.length            
        });`